### ISSUE TITLE
Duplicate rendered characters on next whitespace when using opacity

### ISSUE THREAD

**Issue by alexandrudima** [electron/electron#13367]
* Electron Version: 2.0.2
* Operating System (Platform and Version): macOS 10.13.4
* Last known working Electron version: 1.7.12

This has been noticed when trying to upgrade VS Code to Electron 2.0.2 in https://github.com/Microsoft/vscode/issues/52196

**Expected Behavior**
Characters with opacity in a monospace font should be rendered without duplication (in the next whitespace).

**Actual behavior**
Characters with opacity in a monospace font are rendered doubled when followed by whitespace.

**To Reproduce**
Using https://github.com/electron/fiddle , here is the HTML code needed to reproduce in the renderer:
```html
<!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>Hello World!</title>
  </head>
  <body>

<style>
  .squiggly-inline-unnecessary {
  opacity: 0.6;
}
.view-line {
  font-family: "Monaco";
    font-weight: normal;
    font-size: 14px;
    line-height: 21px;
    letter-spacing: 0px;
}
</style>

<div style="top:21px;height:21px;" class="view-line"><span>
<span class="mtk1">&nbsp;&nbsp;&nbsp;&nbsp;</span>
<span class="mtk5 squiggly-inline-unnecessary">const</span>
<span class="mtk1 squiggly-inline-unnecessary">&nbsp;</span>
<span class="mtk13 squiggly-inline-unnecessary">a</span>
<span class="mtk1 squiggly-inline-unnecessary">&nbsp;</span>
<span class="mtk17 squiggly-inline-unnecessary">=</span>
<span class="mtk1 squiggly-inline-unnecessary">&nbsp;</span>
<span class="mtk9 squiggly-inline-unnecessary">`hdshhdasha;</span>
</span></div>

    <h1>Hello World!</h1>
    <!-- All of the Node.js APIs are available in this renderer process. -->
    We are using Node.js <script>document.write(process.versions.node)</script>,
    Chromium <script>document.write(process.versions.chrome)</script>,
    and Electron <script>document.write(process.versions.electron)</script>.

    <script>
      // You can also require other files to run in this process
      require('./renderer.js')
    </script>
  </body>
</html>
```

The above should render the text:
```
    const a = `hdshhdasha;
```

**Screenshots**

Notice how `a` and `=` are doubled:

![image](https://user-images.githubusercontent.com/9205389/41726379-4994972c-7572-11e8-9063-cda99f13d350.png)

**Additional Information**
This does not reproduce when running with `--disable-gpu`.
This does not reproduce when removing the opacity.
This does not reproduce when using two `&nbsp;`s in a `<span>`.
This does not reproduce on Chrome 67.
This does not reproduce on all macOS machines we have tried, but it reproduces on some Windows machines.
fyi @bpasero @chrmarti

**Comment by bpasero** [comment]
Some findings:
* it does not reproduce when using `will-change:transform` to put that character on a separate layer for rendering
* it does not reproduce with Electron 3.0.x
* I could not find an issue on https://bugs.chromium.org/p/chromium/issues/list regarding the topics "nbsp", "duplicate character", etc :-/

**Comment by bpasero** [comment]
After looking into this with @alexandrudima we think this issue was reported to Chromium via these bugs:
* https://bugs.chromium.org/p/chromium/issues/detail?id=755871
* https://bugs.chromium.org/p/chromium/issues/detail?id=773107
* https://bugs.chromium.org/p/chromium/issues/detail?id=766836

And is fixed via the following commits:
* https://skia-review.googlesource.com/c/skia/+/55381
* https://skia-review.googlesource.com/c/skia/+/55800
* https://skia-review.googlesource.com/c/skia/+/58580

which all circle around a small boolean flag `fUseDrawInsteadOfClear` depending on the GPU configuration.

I assume we cannot backport Skia fixes because Skia seems to be a upstream dependency of Chromium itself, but maybe someone can chime in if this would be feasable?

**Comment by bpasero** [comment]
@sofianguy this also appears to be fixed in Electron 3.0.x.

**Comment by ckerr** [comment]
Closing this ticket as wontfix because backporting a skia update isn't viable.

https://github.com/electron/libchromiumcontent/pull/603#pullrequestreview-138758318

> I think this needs to be discussed with a broader audience before being merged. This brings in around 1,300 new commits which change around 1,000 files, with a total delta of something like 60,000 changed lines. This isn't something to be merged to a stable branch lightly.

https://github.com/electron/libchromiumcontent/pull/603#issuecomment-406591141

> The chromium team advices against this https://groups.google.com/a/chromium.org/d/msg/graphics-dev/_Zi_vxXpf7Q/tov4wNQ7BgAJ Also there were layouttests that got broken with this change.
